src/stan/mcmc/hmc/nuts/base_nuts.hpp

```diff
@@ -249,8 +258,9 @@ namespace stan {
             z_propose = z_propose_right;
         }
 
+        Eigen::VectorXd rho_subtree = rho_left + rho_right;
         rho += rho_subtree;
```

Comment:
rho_subtree is used again in the following call to compute_criterion.  Because of this we'd have to dump it into a temporary or awkwardly call ```.eval()``` anyways.